vyked/decorators/tcp.py

```diff
@@ -86,6 +86,44 @@ def wrapper(self, *args, **kwargs):
     wrapper.is_request = True
     return wrapper
 
+def deprecated_api(func=None, new_api=""):
```

Comment:
DRY. Let's make a separate function for any duplicate code between api and deprecated api instead of copying code.